db/db_impl/db_impl.cc

```diff
@@ -2487,8 +2497,20 @@ Status DBImpl::GetImpl(const ReadOptions& read_options, const Slice& key,
     }
   }
 
+  // RocksDB-Cloud contribution begin
+  auto super_snapshot =
+      dynamic_cast<const SuperSnapshotImpl*>(read_options.snapshot);
+  if (super_snapshot && cfd->GetID() != super_snapshot->cfd()->GetID()) {
+    std::ostringstream oss;
+    oss << "SuperSnapshot column family " << super_snapshot->cfd()->GetName()
+        << " doesn't match provided column family " << cfd->GetName();
+    return Status::InvalidArgument(oss.str());
+  }
+
   // Acquire SuperVersion
-  SuperVersion* sv = GetAndRefSuperVersion(cfd);
+  SuperVersion* sv =
+      super_snapshot ? super_snapshot->sv() : GetAndRefSuperVersion(cfd);
```

Comment:
Does the lack of Ref here mean that we might clean up the SuperVersion while an Iterator is referencing it?

Reply:
No, because the super_snapshot holds a ref to the super version.